### DISCUSSION TITLE
[Help Wanted] [Guide] Translate Part-DB to other languages

### DISCUSSION THREAD

**Discussion by jbtronics** [Part-DB/Part-DB-server#468] [Announcements]
#  Guide: Contribute to Part-DB translations

As you maybe already know Part-DB is available in different languages, to be accessible for all kind of people.
As the maintainer of Part-DB only speaks German and English, other language translations has to be maintained by the community.

Everytime a new piece of text is added to Part-DB it needs to be translated to other languages, or just the english version will be shown.

The translations of Part-DB are managed using Crowdin. If you wanna help Part-DB to be translated to other languages, go to https://part-db.crowdin.com/part-db and login.

You can see the available languages and their progress (how many strings has been translated and how many left untranslated). 
![grafik](https://github.com/Part-DB/Part-DB-server/assets/5410681/d415e5fe-42f3-412a-9d95-ca40f5319357)

If you click on a language you will be able to select a file to edit/view and you will end in an editor, where you can add or change translations:

![grafik](https://github.com/Part-DB/Part-DB-server/assets/5410681/6b6026f1-e9ca-404f-8a43-cc4cef50369a)

Part-DB uses translation keys, so the "source string" does not say much about its meaning. You should expand the "Other Languages" section in the editor, to see the english (and other languages translations). Afterwards click just save and your changes will be stored.

Beware to keep formatting and special symbols in the translations. So if the english translation ends with an dot (".") or contains HTML tags, these should be contained in the translation too. The same goes for placeholders like `%placeholder%`, which get filled with content later.
These should be contained in the translation too at a grammatically reasonable place.

Crowdin gets automatically synced with the Git repository, so any new strings from development will show up on Crowdin and the translations will be get synced back into the git repo later automatically. So you dont have to do anything.
After the pull request was merged in the master branch, you can view Part-DB with the translations in the demo instance, by attaching the two-character language code (e.g. `https://demo.part-db/cs` for the czech version), even if the language is not in the language picker.

So if you know some languages, feel free to go to Crowdin and contribute to the translations. Even if strings are already translated, you can always proofread.

If you want to contribute to a language not already existing at Crowdin, write me a message or post a comment here. I will then add it to Crowdin.

The [CONTRIBUTING.md](https://github.com/Part-DB/Part-DB-server/blob/master/CONTRIBUTING.md) contains additional information about the translation system and specifics.

**Comment by florianhumblot** [comment]
I wouldn't mind looking into adding a Dutch translation if you add it to the options :)

**Reply by jbtronics** [reply]
Okay. I added Dutch as available language.

Good luck.

**Reply by bascuppen** [reply]
@florianhumblot Would you mind me joining aforementioned fun? Or should I ask: "meer zielen, meer vreugd... kan ik je ook helpen met vertalen?"

**Reply by florianhumblot** [reply]
> @florianhumblot Would you mind me joining aforementioned fun? Or should I ask: "meer zielen, meer vreugd... kan ik je ook helpen met vertalen?"
> 

Hey @bascuppen , I really haven't had much time to work on this unfortunately, so any extra help is more than welcome!

**Reply by mvkalsbeek** [reply]
I've done some of the Dutch translation, but there's still a lot left. A review of the translated bits would be welcome as well, mostly for consistency in the translated terms.

**Comment by wombat01** [comment]
Can you please add Czech as well? Thank you.

**Reply by jbtronics** [reply]
I have added it

**Reply by jbtronics** [reply]
@wombat01  Thank you for your very quick translation 👍 
I have merged it into the master branch. You can test it on the demo via: https://demo.part-db.de/cs/

If you think that the translations are fine, I will add czech to the language picker in the navbar.

**Reply by wombat01** [reply]
hi, feel free to add, I'll keep tweaking, depending on what I come up with.

**Comment by vandersonpc** [comment]
Can you please Add Portuguese?

Many thanks!

Vanderson

**Reply by jbtronics** [reply]
Added it

**Comment by jaimelaborda** [comment]
Hi,

I will help to translate to Spanish if you can add it, please.

Many thanks!

Best regards,
Jaime.

**Reply by jbtronics** [reply]
I have added it.

**Comment by grzesiek410** [comment]
Hi,
please add Polish language, I will help with translating.
Thanks

**Reply by jbtronics** [reply]
Added it

**Comment by kgalinec** [comment]
Hi,
can you add Croatian? I can help with translating.
Thanks

**Reply by jbtronics** [reply]
I have added it

**Comment by Datinhkhach** [comment]
Hi

Can you add vietnamese? Thanks

**Reply by jbtronics** [reply]
added it

**Comment by adadnc** [comment]
Turkish is missing, could you add Turkish?

**Reply by jbtronics** [reply]
added it

**Comment by filipecalvo** [comment]
A correction. The portuguese language you added is being translated into Portuguese-BR.

**Comment by RomanOst2020** [comment]
Hi! Please add Ukrainian for translation.
Thank you!

**Reply by jbtronics** [reply]
Done

**Comment by jbtronics** [comment]
If someone wanna use github copilot agent for translations, give me a message and we can try it out.
It seems to work quite well  with translations, but it should still have some human to verify the results...

**Comment by progh2** [comment]
Hello, Add Korean please~ 
Thank you :)

**Reply by jbtronics** [reply]
I added it.

**Reply by progh2** [reply]
I first translated the files locally and tested them on my own Part-DB instance — everything works well.
I've uploaded the Korean translations to Crowdin via the API, so all strings are now translated.
I also reviewed the QA issues that were flagged, but most of them turned out to be false positives (the spell checker doesn't handle Korean compound words and loanwords properly), so no changes were needed.
Here are some screenshots with the Korean translation applied. 
Thank you! ☕

<img width="1161" height="836" alt="image" src="https://github.com/user-attachments/assets/3e277fd4-e415-45ef-8223-61bce5d7414a" />
<img width="1149" height="719" alt="image" src="https://github.com/user-attachments/assets/36182417-1cee-417c-8bcd-7abbf33329b8" />
<img width="1156" height="845" alt="image" src="https://github.com/user-attachments/assets/34f547c2-0194-406f-af1b-bda6a5c21819" />

**Reply by jbtronics** [reply]
Hey thanks. I merged the translations and added korean to the list of selectable languages.

This will be included in the next release and you can check it out in the demo instance.
